domain/application/state/application.go

```diff
@@ -411,184 +411,6 @@ func (st *State) insertCAASApplicationUnits(
 	return nil
 }
 
-// DeleteApplication deletes the specified application, returning an error
-// satisfying [applicationerrors.ApplicationNotFoundError] if the application
-// doesn't exist. If the application still has units, as error satisfying
-// [applicationerrors.ApplicationHasUnits] is returned.
-func (st *State) DeleteApplication(ctx context.Context, name string) error {
```

Comment:
I suspect we'll require this back at some point for model migrations, but we've always got history in any event.